### PR TITLE
add type completer for 'topic pub' and 'service call'

### LINKED ISSUE #63: [ros2 topic pub / ros2 service call] Autocomplete message/service type
#### Feature description

Right now these tools autopopulate the topic / service name based on the discovered topics / services, but dont populate the type.
The structure used (topic/service_names_and_types) also provides the types so we should autocomplete the type matching the selected name.

If no type matching the name is found in the structure we should apply the same logic as the service/message completer used for ros2msg/ros2srv to complete based on the interfaces installed on the system

### PR DESCRIPTION
**Pull request by dirk-thomas**
Connect to #63. Fixes #63.

### CODE REVIEW

**Comment by mikaelarguedas** [comment]
code change looks good overall. Though it would be great to hide the `--repeat` option if only a single service type matches.
e.g.
```
$ ros2 service call /add_two_ints <TAB><TAB> 
example_interfaces/AddTwoInts  -r  --repeat
```

I'd expect this to just complete the matching service name

**Comment by dirk-thomas** [comment]
> Though it would be great to hide the `--repeat` option if only a single service type matches.

I don't think that is technically possible (the results of one completer deciding to ignore the results of other completers). Anyway I wouldn't consider this within the scope of this PR.

**Comment by mikaelarguedas** [comment]
fair enough. Then maybe for another PR we can consider providing the same argument to `ros2 topic pub` ? Because it's a bit surprising from the user side to have a different behavior

**Comment by dirk-thomas** [comment]
> Then maybe for another PR we can consider providing the same argument to `ros2 topic pub` ?

Sure, `topic pub` currently publishes with a fixed rate of once every second. A pull request to make this configurable would be great.

File: ros2service/ros2service/api/__init__.py [hunk 1]
```diff
@@ -43,3 +44,21 @@ def __call__(self, prefix, parsed_args, **kwargs):
                 node=node,
                 include_hidden_services=getattr(
                     parsed_args, self.include_hidden_services_key))
+
+
+class ServiceTypeCompleter:
+    """Callable returning an existing service type or all service types."""
```
Comment (mikaelarguedas) [inline ros2service/ros2service/api/__init__.py]: Nit: "an existing service type" I would recommend the documentation that it's returning the service type matching the passed service_name or all types (same for topics below)

Comment (dirk-thomas) [reply ros2service/ros2service/api/__init__.py]: That sounds good but exceed the one line rule for the docblock :wink:

**Pull request merged**